modules/builtin/src/backend/prompts/enum.ts

```diff
@@ -0,0 +1,104 @@
+import axios from 'axios'
+import { ExtractionResult, IO, Prompt, PromptConfig, ValidationResult } from 'botpress/sdk'
+import * as sdk from 'botpress/sdk'
+import { extractEventCommonArgs } from 'common/action'
+
+import commonFields from './common'
+
+class PromptEnum implements Prompt {
+  private _entity: string
+  private _question: string
+  private _useDropdown: boolean
+
+  constructor({ entity, question, useDropdown }) {
+    this._entity = entity
+    this._question = question
+    this._useDropdown = useDropdown
+  }
+
+  extraction(event: IO.IncomingEvent): ExtractionResult | undefined {
+    const entity = event.nlu?.entities?.find(x => x.type === `custom.list.${this._entity}`)
+    if (entity) {
+      return {
+        value: entity.data.value,
+        confidence: entity.meta.confidence
+      }
+    }
+  }
+
+  async validate(value): Promise<ValidationResult> {
+    if (value == undefined) {
+      return { valid: false, message: 'Provided value is invalid' }
+    }
+
+    return { valid: true }
+  }
+
+  customPrompt = async (event: IO.OutgoingEvent, incomingEvent: IO.IncomingEvent, bp: typeof sdk): Promise<boolean> => {
+    if (!this._useDropdown) {
+      return false
+    }
+
+    const {
+      data: { occurrences }
+    } = await axios.get(
+      `mod/nlu/entities/${this._entity}`,
+      await bp.http.getAxiosConfigForBot(event.botId, { localUrl: true })
+    )
+
+    let payloads = []
+    if (occurrences.length <= 3) {
+      const element = {
+        en: {
+          text: this._question,
+          choices: occurrences.map(x => ({ title: x.name, value: x.name }))
+        }
+      }
```

Comment:
Small helper method to build a full object based on the list of languages for questions

```js
const createMultiLangObject = (questions: { [lang: string]: string }, textProp: string, properties: any) => {
  return Object.keys(questions).reduce((acc, curr) => {
    acc[curr] = {
      [textProp]: questions[curr],
      ...properties
    }
    return acc
  }, {})
}

const element = createMultiLangObject(this._question, 'text', {
      choices: [
        { title: 'Yes', value: 'yes' },
        { title: 'No', value: 'no' }
      ]
    })
```